src/Project/utils.js

```diff
@@ -0,0 +1,95 @@
+import Map from 'ol/Map'
+import LayerVector from 'ol/layer/Vector'
+import GeoJSON from 'ol/format/GeoJSON'
+import olVectorSource from 'ol/source/Vector'
+import olStyle from 'ol/style/Style'
+import olCircleStyle from 'ol/style/Circle'
+import olFill from 'ol/style/Fill'
+import olStroke from 'ol/style/Stroke'
+
+import { loadDataLayer } from 'DataLayers'
+import { VectorLayer } from 'classes'
+import ugh from 'ugh'
+import { version } from '../../package.json'
+
+/**
+ * A utility that takes map state and outputs it as a project file
+ * @function
+ * @category Project
+ * @since 1.9.0
+ * @param {Map} - a reference to openlayers map
+ * @returns {File} - a JSON file in ol-kit project format
+ */
+export async function createProject (map) {
+  if (!(map instanceof Map)) return ugh.throw('\'createProject\' requires a valid openlayers map as the first argument')
+
+  const rawLayers = map.getLayers().getArray()
+  const layers = rawLayers.map(layer => {
+    // some key/values are too large to store in the project file metadata
+    const keysBlacklist = ['source']
+    const keys = layer.getKeys().filter(key => !keysBlacklist.includes(key))
+    const values = {}
+
+    keys.forEach(key => {
+      values[key] = layer.get(key)
+    })
```

Comment:
look at using `.getProperties()` instead

Reply:
it returns the same internal ol values- sticking with original idea